java-frontend/src/main/java/org/sonar/java/xml/maven/PomCheckContextImpl.java

```diff
@@ -67,15 +68,28 @@ public void reportIssue(PomCheck check, int line, String message, List<Location>
   }
 
   @VisibleForTesting
-  static AnalyzerMessage getSecondaryAnalyzerMessage(JavaCheck check, File file, Location location) {
-    XmlLocation startLocation = location.tree.startLocation();
-    XmlLocation endLocation = location.tree.endLocation();
-    TextSpan ts = new TextSpan(
-      startLocation.line(), offsetFromColumn(startLocation.column()),
-      endLocation.line(), offsetFromColumn(endLocation.column()));
+  AnalyzerMessage getSecondaryAnalyzerMessage(JavaCheck check, File file, Location location) {
```

Comment:
nop. let's keep it minimal. I don't want to introduce notion of file+check+analyzerMessage into the location class.